### PR TITLE
feat: build the library with both es5 and es6 as a target

### PR DESCRIPTION
**Pull request by riccardo-forina**
This builds the library targeting both ES5 and ES6 and specifying the right entry points, so that other bundlers can consume the most appropriate version of the library for them.

This should address some issue with tree-shaking and multiple versions of the library being loaded when using uniforms-patternfly from a webpack's federated module.

One error this should fix is the `useForm must be used within a form` one, which is likely to depend on multiple versions of uniform loaded at the same time.

### CODE REVIEW

**Comment by riccardo-forina** [comment]
Sorry for the ping @ljmotta, if you have time to look at this I'd appreciate it 🙏

File: tsconfig.es5.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "baseUrl": "src",
+    "outDir": "es5",
```
Comment (ljmotta) [inline tsconfig.es5.json:5]: Could you add the `es5` inside the `dist` folder?

File: tsconfig.es6.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "baseUrl": "src",
+    "outDir": "es6",
```
Comment (ljmotta) [inline tsconfig.es6.json:5]: Could you add the `es6` inside the `dist` folder?

**Review by ljmotta — COMMENTED** [review]
Hey, thanks for this PR. I was checking it locally, and it works good. I've made just two comments inline, I think it would be better to have the `es5` and `es6` inside the `dist` or even a `lib` folder, just to keep things organized. WDYT?

**Comment by riccardo-forina** [comment]
Sure can, I did it this way just because that's the way uniforms itself handle the build, it seemed a good idea to keep things aligned.

**Review by ljmotta — APPROVED** [review]

**Pull request merged**